### PR TITLE
fix: Issue #446 - End Date datepicker should be initialised to the date of start date

### LINKED ISSUE #446: End Date datepicker should be initialised to the date of start date
**Summary:**

Currently the end date is not set for the datepicker.

**Steps to reproduce:**

Open end date datepicker in any activity.

**Expected behavior:**

The minimum allowed date on the datepicker should be the start date.

**Observed behavior:**

All the dates were allowed in the end date and the error message was only displayed when the form was submitted.

**Device and Android version:**

Samsung A5(2016)

### PR DESCRIPTION
**Pull request by luckyman20**
Fixes #446

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

@therajanmaurya @dilpreet96 Please review it !!

### CODE REVIEW

File: app/src/main/java/org/mifos/mobilebanking/ui/fragments/SavingAccountsTransactionFragment.java [hunk 1]
```diff
@@ -75,11 +75,11 @@
 
     private View rootView;
     private long savingsId;
-    private long startDateFromPicker , endDateFromPicker;
+    public long startDateFromPicker , endDateFromPicker;
```
Comment (dilpreet2028) [inline app/src/main/java/org/mifos/mobilebanking/ui/fragments/SavingAccountsTransactionFragment.java]: Make them private @luckyman20

File: app/src/main/java/org/mifos/mobilebanking/ui/fragments/SavingAccountsTransactionFragment.java [hunk 2]
```diff
@@ -75,11 +75,11 @@
...
     private List<Transactions> transactionsList, dummyTransactionList;
     private SavingsWithAssociations savingsWithAssociations;
-    private DatePick datePick;
-    private DialogFragment mfDatePicker;
+    public DatePick datePick;
```
Comment (dilpreet2028) [inline app/src/main/java/org/mifos/mobilebanking/ui/fragments/SavingAccountsTransactionFragment.java]: Make this private

**Review by dilpreet2028 — CHANGES_REQUESTED** [review]
Good work @luckyman20 , make the changes and also add a gif.

**Comment by luckyman20** [comment]
ezgif com-optimize

@dilpreet96 Done !! :-)

File: app/src/main/java/org/mifos/mobilebanking/utils/MFDatePicker.java [hunk 3]
```diff
@@ -26,6 +26,10 @@
     public static final String TAG = "MFDatePicker";
     static String dateSet;
     static Calendar calendar;
+    public String startDateString;
```
Comment (dilpreet2028) [inline app/src/main/java/org/mifos/mobilebanking/utils/MFDatePicker.java]: Make it private.

File: app/src/main/java/org/mifos/mobilebanking/utils/MFDatePicker.java [hunk 4]
```diff
@@ -26,6 +26,10 @@
...
+    public long startDate;
```
Comment (dilpreet2028) [inline app/src/main/java/org/mifos/mobilebanking/utils/MFDatePicker.java]: Make it private.

File: app/src/main/java/org/mifos/mobilebanking/utils/MFDatePicker.java [hunk 5]
```diff
@@ -26,6 +26,10 @@
...
+    boolean isStart;
```
Comment (dilpreet2028) [inline app/src/main/java/org/mifos/mobilebanking/utils/MFDatePicker.java]: Make it private.

File: app/src/main/java/org/mifos/mobilebanking/utils/MFDatePicker.java [hunk 6]
```diff
@@ -91,5 +102,9 @@ public void setOnDatePickListener(OnDatePickListener onDatePickListener) {
         public void onDatePicked(String date);
     }
 
+    public boolean isStartDate(boolean isStart) {
```
Comment (dilpreet2028) [inline app/src/main/java/org/mifos/mobilebanking/utils/MFDatePicker.java]: As you are using this function to just update the isStart Value so you can change the return type to void and Remove the return statement.

**Review by dilpreet2028 — CHANGES_REQUESTED** [review]
Good work @luckyman20 , but this DatePicker is also being used in other activity i.e. LoanApplicationFragment where we don't have to select a min and a max date, so for that you can initialise isStart = true in the newInstance method as a default.

**Comment by luckyman20** [comment]
@dilpreet96 Made the Changes :-)

**Comment by dilpreet2028** [comment]
Good work @luckyman20

**Review by dilpreet2028 — APPROVED** [review]

**Pull request merged**